andrey/andrey_stored_procedures.sql

```diff
@@ -0,0 +1,55 @@
+-- Write the stored procedure to assign an employee to an existing shift.
+CREATE PROCEDURE uspAssignEmployeeShift
+  @Fname VARCHAR(30),
+  @Lname VARCHAR(30),
+  @NetID VARCHAR(100),
+  @PositionName VARCHAR(50),
+  @ShiftTypeName VARCHAR(50),
+  @LocationName VARCHAR(50),
+  @QuarterName VARCHAR(10),
+  @MonthName VARCHAR(10),
+  @DayName INT,
+  @Hour INT
+  AS
+    DECLARE @EmployeeID INT = (SELECT EmployeeID FROM tblEMPLOYEE WHERE Fname = @Fname AND Lname = @Lname AND UWNetID = @NetID)
+    DECLARE @PositionID INT = (SELECT PositionID FROM tblPOSITION WHERE PositionName = @PositionName)
+    DECLARE @ShiftTypeID INT = (SELECT ShiftTypeID FROM tblSHIFT_TYPE WHERE ShiftTypeName = @ShiftTypeName)
+    DECLARE @LocationID INT = (SELECT LocationID FROM tblLOCATION WHERE LocationName = @LocationName)
+    DECLARE @QuarterID INT = (SELECT QuarterID FROM tblQUARTER WHERE QuartName = @QuarterName)
+    DECLARE @MonthID INT = (SELECT MonthID FROM tblMONTH WHERE MonthName = @MonthName)
+    DECLARE @DayID INT = (SELECT DayID FROM tblDAY WHERE DayName = @DayName)
```

Comment:
Instead of declaring multiple variables, join the respective tables to the larger EmpPos and Shift queries to decrease database calls.

Reply:
I accomplished this by:

```
DECLARE @EmpPosID INT =
      (SELECT EmpPosID
        FROM tblEMPLOYEE_POSITION
        WHERE EmployeeID = (SELECT EmployeeID FROM tblEMPLOYEE WHERE Fname = @Fname AND Lname = @Lname AND UWNetID = @NetID)
        AND PositionID = (SELECT PositionID FROM tblPOSITION WHERE PositionName = @PositionName))
    
    DECLARE @ShiftID INT =
      (SELECT ShiftID
        FROM tblSHIFT
        WHERE LocationID = (SELECT LocationID FROM tblLOCATION WHERE LocationName = @LocationName)
          AND QuarterID = (SELECT QuarterID FROM tblQUARTER WHERE QuartName = @QuarterName)
          AND MonthID = (SELECT MonthID FROM tblMONTH WHERE MonthName = @MonthName)
          AND DayID = (SELECT DayID FROM tblDAY WHERE DayName = @DayName)
          AND ShiftTypeID = (SELECT ShiftTypeID FROM tblSHIFT_TYPE WHERE ShiftTypeName = @ShiftTypeName))
    
    DECLARE @StatusID INT = (SELECT StatusID FROM tblSTATUS WHERE StatusTitle = 'Assigned')

    BEGIN TRANSACTION T1
      INSERT INTO tblEMP_SHIFT_STATUS (StatusID, EmpPosID, ShiftID, DateUpdated)
        VALUES (@StatusID, @EmpPosID, @ShiftID, GETDATE())
    COMMIT TRANSACTION T1
```